app/src/main/java/com/magnox/birthdays/AddOrEditActivity.kt

```diff
@@ -106,42 +109,43 @@ class AddOrEditActivity : AppCompatActivity(), DatePickerDialog.OnDateSetListene
 
         var isValid = true
 
-        if (et_day.text.toString().isEmpty()) {
-            et_day.error = getString(R.string.add_error_field_required)
-            isValid = false
-        }
-        else if (et_day.text.toString().toInt() < 1 || et_day.text.toString().toInt() > 31) {
-            et_day.error = getString(R.string.add_error_invalid_date)
-            isValid = false
-        }
-        else if (et_day.text.toString().isNotEmpty()) {
-            et_day.error = null
-        }
-
         if (et_month.text.toString().isEmpty()) {
             et_month.error = getString(R.string.add_error_field_required)
             isValid = false
-        }
-        else if (et_month.text.toString().toInt() < 1 || et_month.text.toString().toInt() > 12) {
+        } else if (et_month.text.toString().toInt() < 1 || et_month.text.toString().toInt() > 12) {
             et_month.error = getString(R.string.add_error_invalid_date)
             isValid = false
-        }
-        else if (et_month.text.toString().isNotEmpty()) {
+        } else if (et_month.text.toString().isNotEmpty()) {
             et_month.error = null
         }
 
         if (et_year.text.toString().isEmpty()) {
             et_year.error = getString(R.string.add_error_field_required)
             isValid = false
-        }
-        else if (et_year.text.toString().toInt() < 1 || et_year.text.toString().toInt() > 10000) {
+        } else if (et_year.text.toString().toInt() < 1 || et_year.text.toString().toInt() > 10000) {
             et_year.error = getString(R.string.add_error_invalid_date)
             isValid = false
-        }
-        else if (et_year.text.toString().isNotEmpty()) {
+        } else if (et_year.text.toString().isNotEmpty()) {
             et_year.error = null
         }
 
+        var daysInMonth = 31
+        if (isValid) {
+            val yearMonthObject =
+                YearMonth.of(et_year.text.toString().toInt(), et_month.text.toString().toInt())
+            daysInMonth = yearMonthObject.lengthOfMonth()
+        }
```

Comment:
why use `isValid` check? day can be invalid when month was invalid as well

Reply:
The validation for the day is some lines later. At this place there is only calculated how many days this specific month has, and this is only possible if there is a valid year and month

Reply:
got it!